tap_freshdesk/client.py

```diff
@@ -183,6 +186,48 @@ def _wait_for(exception) -> int:
     
     def backoff_jitter(self, value: float) -> float:
         return value
+    
+    # Handling error, overriding this method over RESTStream's Class
+    def response_error_message(self, response: requests.Response) -> str:
+        """Build error message for invalid http statuses.
+
+        WARNING - Override this method when the URL path may contain secrets or PII
+
+        Args:
+            response: A :class:`requests.Response` object.
+
+        Returns:
+            str: The error message
+        """
+        full_path = urlparse(response.url).path or self.path
+        error_type = (
+            "Client"
+            if HTTPStatus.BAD_REQUEST
+            <= response.status_code
+            < HTTPStatus.INTERNAL_SERVER_ERROR
+            else "Server"
+        )
+
+        error_details = []
+        if response.status_code >= 400:
+            print(f"Error Response: {response.status_code} {response.reason}")
+            try:
+                error_data = response.json()
+                errors = error_data.get('errors')
+                index = 0
+                for error in errors:
+                    message = error.get('message', 'Unknown')
+                    field = error.get('field', 'Unknown')
+                    error_details.append(f"Error {index + 1}: Message - {message}, Field - {field}")
+                    index += 1
+            except json.JSONDecodeError:
```

Comment:
I think this needs to be `requests.exceptions.JSONDecodeError` not `json.JSONDecodeError`.

```
requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

Reply:
Got it, this has been implemented using `requests.exceptions.JSONDecodeError`